### ISSUE TITLE
Allow explanation of eslint-disable

### ISSUE THREAD

**Issue by bfred-it** [eslint/eslint#11806]
**The version of ESLint you are using.**
eslint@5.16.0

**The problem you want to solve.**
I want to disable a feature but I need to explain why.

This can be done on a separate line when you use block-level `eslint-disable` but it becomes awkward when using one-line disablers like `eslint-disable-next-line`.

**Your take on the correct solution to problem.**

```js
// eslint-disable-next-line your-rule because "explanation"
// eslint-disable-next-line your-rule, "explanation"
// eslint-disable-next-line your-rule // "explanation"
// eslint-disable-next-line your-rule - "explanation"
```

**Are you willing to submit a pull request to implement this change?**

Way out of my comfort zone

**Comment by kaicataldo** [comment]
Thanks for the proposal.

Do you mind expanding on why a separate explanation line comment either on the offending line or above the line that contains the disable comment won’t suffice in your case?

I personally am not not convinced we should make this change, as allowing for arbitrary characters in disable comments adds a lot of complexity and surface area for bugs for what seems like very little gain. Adding the ability to do this feels like a mixing of responsibilities of comments that are part of ESLint’s API and comments that are intended for other humans as well as creating a more complex/muddied API.

Let’s see what others have to say.

**Comment by kaicataldo** [comment]
Additionally, while you’re more than welcome to discuss and get feedback here first, our policy is that any change to core requires creating an [RFC](https://github.com/eslint/rfcs). Thanks!

**Comment by bfred-it** [comment]
I'll use a real-life example.

## Ideal (same line)

The comment is exactly where it needs to be: explaining a line, on the line itself

```
// eslint-disable-next-line @typescript-eslint/no-extraneous-class // Because it follows `@types/chrome`'s style"
export class RequestContentScript {
	constructor(contentScript: ContentScript);
}
```

## Good (next line, not allowed with `next-line`)

The comment follows the thought "This is disabled \n This is why"

```
// eslint-disable-next-line @typescript-eslint/no-extraneous-class
// Because it follows `@types/chrome`'s style"
export class RequestContentScript {
    constructor(contentScript: ContentScript);
}
```

## Currently

"This is why, I do this" seems backwards. "This" what?

```
// This follows `@types/chrome`'s style"
// eslint-disable-next-line @typescript-eslint/no-extraneous-class
export class RequestContentScript {
    constructor(contentScript: ContentScript);
}
```

Making it more clear requires making it wordy

```
// The rule can be disabled because the class follows `@types/chrome`'s style"
// eslint-disable-next-line @typescript-eslint/no-extraneous-class
export class RequestContentScript {
    constructor(contentScript: ContentScript);
}
```

**Comment by glen-84** [comment]
I was also looking for this.

With [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer), you can use `--`, for example:

```php
// phpcs:disable PEAR,Squiz.Arrays -- this isn't our code
```

I used the same for TSLint (although there was [a regression](https://github.com/palantir/tslint/issues/1484) for "disable all" comments).

How are these comments currently processed? With regular expressions?

**Comment by glen-84** [comment]
@mysticatea Should this issue be reopened now that the RFC has been merged?

**Comment by mysticatea** [comment]
> Once an RFC is merged into this repo, then the authors may implement it and submit a pull request to the appropriate ESLint repo without opening an issue.
>
> https://github.com/eslint/rfcs#the-rfc-lifecycle

Therefore, I think that we don't need to reopen.

**Comment by glen-84** [comment]
That's fine if you plan to do it in the very near future, but if you don't, then we don't have a way of voting on and tracking progress of the issue.

**Comment by mysticatea** [comment]
Because it's a breaking change, it will be implemented in the next major release.
